osbs/repo_utils.py

```diff
@@ -60,10 +60,31 @@ def __init__(self, dir_path='', file_name=REPO_CONFIG_FILE):
             with open(file_path) as f:
                 self.container = (yaml.load(f) or {})
 
+        modules = self.container.get('compose', {}).get('modules', [])
+        self.container_module_specs = [ModuleSpec.from_text(module) for module in modules]
+
     def is_autorebuild_enabled(self):
         return self._config_parser.getboolean('autorebuild', 'enabled')
 
 
+class ModuleSpec(object):
+    __slots__ = ('name', 'stream', 'version')
+
+    def __init__(self, name, stream, version=None):
+        self.name = name
+        self.stream = stream
+        self.version = version
+
+    @classmethod
+    def from_text(cls, text):
+        pieces = text.split(':')
+        if not 1 < len(pieces) < 4:
+            raise ValueError('Module specification should be NAME:STREAM or NAME:STREAM:VERSION')
+        if any(not len(p) for p in pieces):
```

Comment:
non-emptiness in python is tested by `not p`.

in your case you can use `not all(pieces)` to make it more readable
```
>>> l = ['something', 'something', '']
>>> all(l)
False
```

Reply:
s/non-emptiness/emptiness/